File: 1.1/src/cz/mg/vulkan/PFNvkCmdDraw.java
package cz.mg.vulkan;

public class PFNvkCmdDraw extends VkFunctionPointer {
    public PFNvkCmdDraw() {
    }

    protected PFNvkCmdDraw(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdDraw(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdDraw(long value) {
        setValue(value);
    }

    public PFNvkCmdDraw(VkInstance instance) {
        super(instance, new VkString("vkCmdDraw"));
    }

    public void call(VkCommandBuffer commandBuffer, int vertexCount, int instanceCount, int firstVertex, int firstInstance){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), vertexCount, instanceCount, firstVertex, firstInstance);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, int vertexCount, int instanceCount, int firstVertex, int firstInstance);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdDraw.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdDraw_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint vertexCount, jint instanceCount, jint firstVertex, jint firstInstance){
    (void)env;
    (void)jc;
    PFN_vkCmdDraw f = (PFN_vkCmdDraw)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (uint32_t)vertexCount,
        (uint32_t)instanceCount,
        (uint32_t)firstVertex,
        (uint32_t)firstInstance
    );
}

